automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/WorkspaceNotebooksPage.scala

```diff
@@ -11,10 +11,14 @@ import org.scalatest.selenium.Page
 class WorkspaceNotebooksPage(namespace: String, name: String)(implicit webDriver: WebDriver)
   extends WorkspacePage(namespace, name) with Page with PageUtil[WorkspaceNotebooksPage] {
 
-  override def awaitReady(): Unit = await enabled sparkClustersHeader
+  override def awaitReady(): Unit = {
+    await condition { clustersTable.getState == "ready" || find(clusterErrorMessage).isDefined }
```

Comment:
I wish this could just delegate to `Table.awaitReady` but I see you're also checking for an error case. While this is okay, I'm interested in hearing ideas for improvements.

Reply:
Eventually, when we bring notebooks out from under the whitelist, it will not longer check for an error case (since everyone will be able to see notebooks) - so this is more of an interim solution. 

Looking at my test failures though - removing awaitReady and doing getState actually makes this line not wait for the table to appear - since awaitReady actually does an `await condition` itself. Calling getState directly tries to immediately look for the table. So I can't do that. :(

what if instead I did
`await condition { find(clustersTable).isDefined || find(clusterErrorMessage).isDefined }`
with `clustersTable` being a testId rather than a Table?

Reply:
Checking for ready state is almost certainly better than just checking if the element exists. It's fine as it. It can be revisited when the error case goes away.

Reply:
The tests aren't passing  with `clustersTable.getState == "ready"` though, since it's just checking immediately, so we can't actually do it that way